src/main/scala/com/dwolla/lambda/cloudflare/requests/processors/RateLimitProcessor.scala

```diff
@@ -0,0 +1,104 @@
+
+  import RateLimitProcessor._
+
+  def this(executor: StreamingCloudflareApiExecutor[F]) = this(ZoneClient(executor), RateLimitClient(executor))
+
+  override def process(action: CloudFormationRequestType,
+                       physicalResourceId: Option[PhysicalResourceId],
+                       properties: JsonObject): Stream[F, HandlerResponse] =
+    for {
+      request <- parseRecordFrom[WrappedRateLimit](properties, "RateLimit")
+      resp <- handleAction(action, request, physicalResourceId, properties)
+    } yield resp
+
+  private def handleAction(action: CloudFormationRequestType,
+                           request: RateLimit,
+                           physicalResourceId: Option[PhysicalResourceId],
+                           properties: JsonObject,
+                          ): Stream[F, HandlerResponse] = (action, physicalResourceId) match {
+    case (CreateRequest, None) => handleCreate(request, properties)
+    case (UpdateRequest, PhysicalResourceId(zid, prid)) => handleUpdate(zid, prid, request)
+    case (DeleteRequest, PhysicalResourceId(zid, prid)) => handleDelete(zid, prid)
+    case (CreateRequest, Some(id)) => Stream.raiseError(UnexpectedPhysicalId(id))
+    case (_, Some(id)) => Stream.raiseError(InvalidCloudflareUri(id))
+    case (UpdateRequest, None) | (DeleteRequest, None) => Stream.raiseError(MissingPhysicalId(action))
+    case (OtherRequestType(_), _) ⇒ Stream.raiseError(UnsupportedRequestType(action))
+  }
+
+  private def handleCreate(request: RateLimit, properties: JsonObject): Stream[F, HandlerResponse] =
+    for {
+      zone <- parseRecordFrom[String](properties, "Zone")
+      zoneId <- zoneClient.getZoneId(zone)
+      resp <- rateLimitClient.create(zoneId, request)
+    } yield HandlerResponse(tagPhysicalResourceId(resp.id.fold("Unknown RateLimit ID")(id => rateLimitClient.buildUri(zoneId, id))), JsonObject(
+      "created" -> resp.asJson
+    ))
+
+  private def handleUpdate(zoneId: ZoneId, rateLimitId: RateLimitId, request: RateLimit): Stream[F, HandlerResponse] =
+    for {
+      resp <- rateLimitClient.update(zoneId, request.copy(id = Option(rateLimitId)))
+    } yield HandlerResponse(tagPhysicalResourceId(rateLimitClient.buildUri(zoneId, resp.id.getOrElse(rateLimitId))), JsonObject(
+      "updated" -> resp.asJson
+    ))
+
+  private def handleDelete(zoneId: ZoneId, rateLimitId: RateLimitId): Stream[F, HandlerResponse] =
+    for {
+      resp <- rateLimitClient.delete(zoneId, rateLimitId)
+    } yield HandlerResponse(tagPhysicalResourceId(rateLimitClient.buildUri(zoneId, resp)), JsonObject(
+      "deleted" -> resp.asJson
+    ))
+
+  private object PhysicalResourceId {
+    def unapply(arg: Option[PhysicalResourceId]): Option[(ZoneId, RateLimitId)] =
+      arg.flatMap(rateLimitClient.parseUri)
+  }
+
+  object RateLimitProcessor {
+
+    case class WrappedRateLimit(rateLimit: RateLimit)
```

Comment:
I wasn't completely happy with the name I ended up choosing here. I'm open to alternatives.